internal/component/otelcol/exporter/kafka/kafka.go

```diff
@@ -49,23 +66,68 @@ type Arguments struct {
 	Brokers                              []string      `alloy:"brokers,attr,optional"`
 	ResolveCanonicalBootstrapServersOnly bool          `alloy:"resolve_canonical_bootstrap_servers_only,attr,optional"`
 	ClientID                             string        `alloy:"client_id,attr,optional"`
-	Topic                                string        `alloy:"topic,attr,optional"`
+	Topic                                string        `alloy:"topic,attr,optional"` // Deprecated
 	TopicFromAttribute                   string        `alloy:"topic_from_attribute,attr,optional"`
-	Encoding                             string        `alloy:"encoding,attr,optional"`
+	Encoding                             string        `alloy:"encoding,attr,optional"` // Deprecated
 	PartitionTracesByID                  bool          `alloy:"partition_traces_by_id,attr,optional"`
 	PartitionMetricsByResourceAttributes bool          `alloy:"partition_metrics_by_resource_attributes,attr,optional"`
 	Timeout                              time.Duration `alloy:"timeout,attr,optional"`
 
+	Logs    *KafkaExporterSignalConfig `alloy:"logs,block,optional"`
+	Metrics *KafkaExporterSignalConfig `alloy:"metrics,block,optional"`
+	Traces  *KafkaExporterSignalConfig `alloy:"traces,block,optional"`
+
 	Authentication otelcol.KafkaAuthenticationArguments `alloy:"authentication,block,optional"`
 	Metadata       otelcol.KafkaMetadataArguments       `alloy:"metadata,block,optional"`
 	Retry          otelcol.RetryArguments               `alloy:"retry_on_failure,block,optional"`
 	Queue          otelcol.QueueArguments               `alloy:"sending_queue,block,optional"`
 	Producer       Producer                             `alloy:"producer,block,optional"`
+	TLS            *otelcol.TLSClientArguments          `alloy:"tls,block,optional"`
 
 	// DebugMetrics configures component internal metrics. Optional.
 	DebugMetrics otelcolCfg.DebugMetricsArguments `alloy:"debug_metrics,block,optional"`
 }
 
+type KafkaExporterSignalConfig struct {
+	Topic    string `alloy:"topic,attr,optional"`
+	Encoding string `alloy:"encoding,attr,optional"`
+}
+
+type deprecatedAttr struct {
+	value        string
+	defaultValue string
+}
+
+func (c *KafkaExporterSignalConfig) convert(topic, encoding deprecatedAttr) kafkaexporter.SignalConfig {
+	result := kafkaexporter.SignalConfig{}
+
+	if c != nil { // Use values from the new block if set.
+		if len(c.Topic) > 0 {
+			result.Topic = c.Topic
+		}
+		if len(c.Encoding) > 0 {
+			result.Encoding = c.Encoding
+		}
```

Comment:
If the block is not nil, should we pass through and potentially use the deprecated field defaults?

Reply:
No. The intent here is to always use the block's explicit or default values if the block is explicitly defined. That way we could document it in a simple way - if the block exists, use the block values. It shouldn't matter if the defaults or if the explicit values are used.

Reply:
The current behavior appears to fall through and use the default value of the deprecated fields if the provided values are len 0

Reply:
Ah I see! Those are not the deprecated field defaults. Those are the defaults to use if neither the deprecated nor the "new" field has a non-empty value. I added comments to make this more clear. The confusion comes from the name of the `deprecatedArg` struct... but I don't know what else to call it?